bites.py

```diff
@@ -275,3 +286,81 @@ def print_anthill_food_tokens(self):
   def print_wine_rule_statement(self):
     print("\nThe wine scoring card currently in play is: ")
     print("%s" % self.wine_rule.capitalize())
+
+  def identify_chocolate_limit(self, partial_trail):
+    """Find the lower limit of where chocolate could be placed into the trail. 
+
+    Parameters
+    ----------
+    partial_trail : (list)
+      A list of tokens that will be used as the trail for this game. 
+      Elements are standard food tokens and wine as strings.
+
+    Returns
+    -------
+    chocolate_limit : (int)
+      The index within the trail that is the first position chocolate is 
+      allowed to be placed.
+    """
+    chocolate_limit = max([partial_trail.index(food) for food in self.standard_tokens_for_trail]) + 2
+    return chocolate_limit
+
+  def add_chocolate_into_trail(self, partial_trail, chocolate_tokens_for_trail):
+    """Adds chocolate tokens into the trail as appropriate
+
+    Shuffles the given number of chocolate tokens into the slice of the trail 
+    after the chocolate limit.
+
+    Parameters
+    ----------
+    partial_trail : (list)
+      A shuffled list of tokens other than chocolate that will be used for this game.
+      Elements are standard food or wine tokens as strings
+    
+    chocolate_tokens_for_trail : (dict)
+      A dictionary containing the chocolate tokens that will be used for this game.
+      Key is 'chocolate' as string
+      Value is int
+
+    Returns
+    -------
+    trail : (list)
+      A shuffled list of all tokens that will be used for this game. 
+      The slice pre-chocolate_lim is unchanged.
+      The slice post-chocolate_limit has had chocolate added and been reshuffled.
+      Elements are standard, wine and chocolate tokens as strings.
+    """
+    trail = [] 
+    for food, amount in chocolate_tokens_for_trail.items():
+      trail = partial_trail + [food] * amount
+      random.shuffle(trail[self.identify_chocolate_limit(partial_trail):])
```

Comment:
I would like to see an integration test for `add_chocolate_into_trail()` when you're not mocking the shuffle and you're somehow checking that it has actually been shuffled. Because I don't think this line is going to work. 
![image](https://user-images.githubusercontent.com/18528532/99828756-467fe980-2b53-11eb-8050-72f12731a0b6.png)